lua/apisix/http/balancer.lua

```diff
@@ -23,12 +25,46 @@ local _M = {
 }
 
 
+local function parse_addr(addr)
+    local pos = find_str(addr, ":", 1, true)
```

Comment:
how about use `ngx.re.split`, which more readable.

Reply:
We need a simple implementation and the current implementation is good.